src/types/person.rs

```diff
@@ -21,34 +22,59 @@ pub struct Person {
 }
 
 impl Person {
-    /// Constructs a new person from a chunk vector according to the specs of
-    /// [Nicolas Markey in "Tame the BeaST"][taming], pp. 23-24.
+    /// Constructs a new person from a chunk vector:
+    /// 1. according to the specs of
+    ///     [Nicolas Markey in "Tame the BeaST"][taming], pp. 23-24.
+    /// 2. biblatex extended name format according to the
+    ///     [documentation of biblatex][biblatex], section 3.4 pp. 80-81,
+    ///     and section §4.2.3 pp. 164-165.
+    ///     Support is limited to default `nameparts`: prefix, family,
+    ///     suffix, given.
     ///
     /// [taming]: https://ftp.rrze.uni-erlangen.de/ctan/info/bibtex/tamethebeast/ttb_en.pdf
+    /// [biblatex]: https://ctan.gutenberg-asso.fr/macros/latex/contrib/biblatex/doc/biblatex.pdf
     pub fn parse(chunks: ChunksRef) -> Self {
-        let num_commas: usize =
-            chunks
-                .iter()
-                .map(|val| {
-                    if let Chunk::Normal(s) = &val.v {
-                        s.matches(',').count()
-                    } else {
-                        0
-                    }
-                })
-                .sum();
+        // Extended Name Format
+        let num_equals = count_num_char(chunks, '=');
+        if num_equals > 0 {
+            return Self::parse_extended_name_format(chunks);
+        }
+        // BibTex
+        Self::parse_bibtex(chunks)
+    }
+
+    /// Constructs new person from chunk slices.
+    fn parse_extended_name_format(chunks: ChunksRef) -> Self {
+        let mut person = HashMap::new();
+        for chunk in split_token_lists(chunks, ",") {
+            let (key, value) = split_at_normal_char(&chunk, '=', true);
+            let key = key.format_verbatim();
+            let value = value.format_verbatim();
+            person.insert(key, value);
+        }
+
+        let name = person.get("family").unwrap_or(&String::new()).to_string();
+        let given_name = person.get("given").unwrap_or(&String::new()).to_string();
+        let prefix = person.get("prefix").unwrap_or(&String::new()).to_string();
+        let suffix = person.get("suffix").unwrap_or(&String::new()).to_string();
```

Comment:
```suggestion
        let name = person.get("family").unwrap_or_default().to_string();
        let given_name = person.get("given").unwrap_or_default().to_string();
        let prefix = person.get("prefix").unwrap_or_default().to_string();
        let suffix = person.get("suffix").unwrap_or_default().to_string();
```

Reply:
Not sure why, but for this particular part I get
```rust
error[E0277]: the trait bound `&String: Default` is not satisfied
    --> src/types/person.rs:57:41
     |
57   |         let name = person.get("family").unwrap_or_default().to_string();
     |                                         ^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `&String`
     |
     = help: the trait `Default` is implemented for `String`
note: required by a bound in `Option::<T>::unwrap_or_default`
    --> /home/quachpas/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:1006:12
     |
1004 |     pub fn unwrap_or_default(self) -> T
     |            ----------------- required by a bound in this associated function
1005 |     where
1006 |         T: Default,
     |            ^^^^^^^ required by this bound in `Option::<T>::unwrap_or_default`
```

Reply:
I'm sorry to have suggested something that breaks. This should work instead: `person.get("family").map(ToString::to_string).unwrap_or_default()`

Reply:
It's fine! This one does work, thank you!